src/connection.ts

```diff
@@ -49,6 +54,7 @@ export const SFDX_HTTP_HEADERS = {
 };
 
 export const DNS_ERROR_NAME = 'Domain Not Found';
+type deployOptions = DeployOptions & { rest?: boolean };
```

Comment:
You should always export types used in public methods, maybe `DeployOptionsWithRest`?